### ISSUE TITLE
rule: no-k8s-kind-secret

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#112]
Kubernetes's `Kind: Secret` includes credentials as plain format.
It just encode value with base64.

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
data:
  username: YWRtaW4=
  password: MWYyZDFlMmU2N2Rm
```

So, `Kind: Secret` manifest should not be commited into a repository.

We can create a rule that disallow to use `Kind: Secret`.
It help you to prevent commit credentials.

* [Secrets - Kubernetes](https://kubernetes.io/docs/concepts/configuration/secret/)
* [GitOps - Frequently Asked Questions](https://www.weave.works/technologies/gitops-frequently-asked-questions/#manage-secrets)
* [Distribute Credentials Securely Using Secrets - Kubernetes](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/)
* [How to keep your Kubernetes secrets secure in Git](https://learnk8s.io/kubernetes-secrets-in-git)

In GitOps context, use another solution like SealedSecret, Vault etc...

- [Secret Management - Argo CD - Declarative GitOps CD for Kubernetes](https://argoproj.github.io/argo-cd/operator-manual/secret-management/)

**Comment by azu** [comment]
This is simiar concet with `.env` #66
